### ISSUE TITLE
Works poorly or not at all in Visual Studio 2022 Enterprise

### ISSUE THREAD

**Issue by DavidPressman** [JoacimWall/Visual-studio-multilingual-extension#10]
Current instruction video is a couple of years old, only illustrates Mac use, and several of its instructions did not match what I saw in VS 2022 Windows.

Below describes my experiences using the latest VS 2022 that was current as of October 15, 2022.

Windows:
1. Tried to use in VS 2022 Windows with my existing large multi-project Maui Blazor project.
The auto-generated .json settings file got placed in the same folder as the solution, not in the desired project folder. (I copied it into that folder just to be sure.)
While "Multilingual Settings" does get listed in VS's "Tools" drop-down, clicking on it has no discernible effect.
So I edited the .json file manually (and later copied to Mac as described below.)
The only function I could get to work was syncing a French resx file with the master English.
Both files then had hundreds of identical entries.
"Translate all files" had no discernible effect, nor was the "Translate this file" choice available when the French file was selected.

2. Created a test blank Maui Blazor single project solution with a master resx file whose English text was just these two words (without the quotes) separated by a carriage return:
"Read
Instructions" 
Created and synced a French version.
"Translate this file" was still unavailable for the French version.
Using the Microsoft translator the first word got translated, but the second simply replaced with "/r/n"

In short, the extension does not perform usefully in VS 2022 for Windows, at least in a Maui project.

Mac:
Created a test blank Xamarin iOS project with a master resx file whose English text was identical to my Windows test.
"Multilingual Settings" does NOT get listed in VS's "Tools" drop-down, so used .json file copied from Windows.
Created and synced a French version,.
"Translate this file" was available for the French version; I used that.
Using the Microsoft translator the first word got translated, but not the second, which remained unchanged.
Using the Google translator the first word was translated but the second simply replaced with "/r/n".
(The first word's Google translation did not match the translation I could create with Google Translate in a browser.)
In short, the extension does not perform usefully in VS 2022 for Mac, at least for Xamarin..

Also needed: While translation is occurring, some visual indication. For very long files there is no other way to know when translation has completed.

**Comment by JoacimWall** [comment]
Hi 
Thanks for you feedback. 

My goal with this extension is to share as much code as possible between Mac and Windows. Unfortunately, Microsoft has no documentation at all regarding extensions for visual studio for Mac, so it makes it a bit difficult.

Have you tested the two sample projects?
https://github.com/JoacimWall/Visual-studio-multilingual-extension/tree/main/src/Samples

or

Can you please submit the project that not working so I can debug and fix if there is any error in my code. 
if you dont want to share all code just share the resx files and the settings.json file and what type of solution.  

1. Regarding the settings menu it should now be removed in the GUI.  
2. Regarding the settings.json file. The file is in the solution folder. (Before it was project folder on mac and solution folder on windows) 
3. Regarding progress for sync, translate, Export and Import now you will see this in a output windows on the Windows platform.  
![SyncLog](https://user-images.githubusercontent.com/9278908/196201041-2aab9b3b-4b6b-4beb-b796-ff3168657ccf.png)

I have also fix same bugs in this release 2.0.8

**Comment by DavidPressman** [comment]
Thank you for your latest update and for your rapid response.
I have not yet tried (but may soon) either sample as neither (Console or UWP) falls within my current area of concern, Maui Blazor.

I am attaching a copy my Maui Blazor Test solution, Windows version.

After updating your extension, I did note each change you described occurs. Good!
I understand the reason for having the .json in the solution folder itself, although that means it is not immediately visible in Solution Explorer and instead is easiest edited with a separate text editor.

I added into Test attempts to translate the words Instruction, Instructions, Message, and Messages -- each both on single lines and on the second of two lines. The attached French file shows results of Microsoft Translate -- none of those common words were rendered into French, although the word Read always was. (I've of course removed my Microsoft key in the attached.)

After preparing the attached, I returned to the original and switched to Google Translate, with nearly identical results on the single line occurrences
(Exception: "Instructions" got rendered as "Des instructions".)

Since neither Microsoft nor Google translated those 4 words, the problem would appear to be in the extension, not those two services.

In all two-line Google cases the second line got rendered only as "/r/n".

Now that Multilingual App Toolkit finally "works" with Maui Blazor (with varying degrees of success on the 13 languages I have so far tried in my main solution), that becomes my preference for now.

Nonetheless, I certainly appreciate your efforts and desire to aid us developers. I wish you the best.
If, in time, your extension proves more reliable than the Toolkit, I will switch.

David Pressman


________________________________

**Comment by JoacimWall** [comment]
Hi David I can't se the attached  "Maui Blazor Test solution". 

Can you please provide it so I can try to fix the issue. 
Are you using resx or resw files?

Thanks Joacim 
Joacim.wall@gmail.com

**Comment by JoacimWall** [comment]
Hi 
I have now fix the multilane error in translation for google. 
I have also tested this on microsoft  translation and it works. 
{
  "MasterLanguageCode": "en-US",
  "TranslationService": 2,
  "ExportFileType": 2,
  "ExportMasterFileOnExport": false,
  "TranslationServiceMsoftEndpoint": "https://api.cognitive.microsofttranslator.com/",
  "TranslationServiceMsoftLocation": "northeurope",
  "TranslationServiceMsoftKey": "xxxxxxxxxx"
}

I have also added a MAUI Project to the sample collection where I have tested that translations of resx file works om Mac and Windows.
There is a new version 2.0.9 Where this is corrected. 

//Joacim
